### ISSUE TITLE
Very different clusters and UMAP from SCT transformed integration datasets

### ISSUE THREAD

**Issue by yzhjh** [satijalab/seurat#8648]
**Hi all,

I've been trying this new Seurat v5 recently and was following the guidelines on the webpage. However, I faced issues regarding integrating the SCT-transformed datasets.

So, using the pipeline for SCT transformed integration in the previous Seurat version, I could get a good clustering UMAP as shown here:**

![Rplot01](https://github.com/satijalab/seurat/assets/164055036/241bac2d-3744-4fce-a9ec-4547032d8e0d)

**However, if I used the new Seurat v5 SCT integration streamline pipeline with similar parameters, I got way more clusters (over 80) and more random scatters instead of fine-defining clusters. The UMAP looks like this:**

![Rplot](https://github.com/satijalab/seurat/assets/164055036/ff12d4e7-22f8-4651-ad0f-2174085b1a7f)

**This discrepancy doesn't happen if I'm integrating datasets without SCT transformation. So, I really want to know what I did not do right in the pipeline I used.

I would really appreciate any comments or help!

The codes for the v4 SCT transformed integration are:**

scrna.list <- SplitObject(scrna, split.by = "DataSet")
scrna.list <- lapply(X = scrna.list, FUN = SCTransform, vars.to.regress = c("percent.mito"))
features <- SelectIntegrationFeatures(object.list = scrna.list, nfeatures = 3000)
scrna.list <- PrepSCTIntegration(object.list = scrna.list, anchor.features = features)
anchor <- FindIntegrationAnchors(object.list = scrna.list, normalization.method = "SCT", anchor.features = features)
scrna <- IntegrateData(anchorset = anchor, normalization.method = "SCT")
scrna <- RunPCA(scrna, verbose = FALSE)
scrna <- RunUMAP(scrna, reduction = "pca", dims = 1: 23)
scrna <- FindNeighbors(object=scrna, dims=1: 23);
scrna <- FindClusters(object=scrna, resolution= 0.3);
DimPlot(object = scrna, reduction = "umap", group.by = "seurat_clusters", pt.size=0.2, label = T)

**The codes for the v5 SCT transformed integration are:**

options(future.globals.maxSize = 1e+10)
scrna <- SCTransform(scrna, vst.flavor = 'v2')
scrna <- IntegrateLayers(object = scrna, method = CCAIntegration, normalization.method = "SCT", verbose=T, assay = "SCT")
scrna <- FindNeighbors(scrna, reduction = "integrated.dr", dims = 23)
scrna <- FindClusters(scrna, resolution = 0.3)
scrna <- RunUMAP(scrna, reduction = "integrated.dr", dims = 1:23)
DimPlot(scrna, reduction = "umap", group.by = c("DataSet","seurat_clusters"))

**Comment by jdrnevich** [comment]
Are the codes above exactly what you ran? Because the v4 has:
`scrna <- FindNeighbors(object=scrna, dims=1: 23);`
while the v5 is only using dim 23 instead of 1:23:
`scrna <- FindNeighbors(scrna, reduction = "integrated.dr", dims = 23)`

**Comment by yzhjh** [comment]
Thank you @jdrnevich! That solved the problem. That might be a typo when I was trying to redo this analysis on Seurat v5. 

I do, have a follow-up question though. Since I was looking into the difference between v4 and v5, I saw v4 uses integrateData() and v5 uses integrateEmbeddings() as a final step in data integration. I looked into the document but could not understand whether that would lead to different results. 

Again, really appreciate you pointing out that typo in my code and I should've been more careful when I tried to debug...
